### DISCUSSION TITLE
v0.28.1

### DISCUSSION THREAD

**Discussion by ramfox** [n0-computer/iroh#2902] [Announcements]
## 🛑 deprecations
  - `iroh-blobs`, `iroh-docs`, `iroh-gossip` will not longer be re-exported in iroh and must be added as their own crates by v0.29.0
  - `Node::client` is deprecated and will be removed in v0.29.0
  - the iroh CLI is deprecated and will be removed in v0.29.0

## ⚠️ Breaking Change
- `iroh-net`
  - removed 
    - the `secret_key` property of the `iroh-relay` config
    - `iroh_net::relay::server::ServerActorTask::secret_key`
    - `iroh_net::relay::server::ServerActorTask::public_key`
    - `iroh_net::relay::server::ServerActorTask::meta_cert`
    - field `secret_key` in `iroh_net::relay::server::RelayConfig`
    - `iroh_net::portmapper` is moved to its own crate: `portmapper`
    - `iroh_net::net` has been split and added to the `netwatch` crate
  - changed
    - `iroh_net::relay::server::ServerActorTask::new` now takes no arguments
    - `MagicsockMetrics` is now `#[non_exhaustive]`. This allows us to add more metrics without breaking backwards compatibility in the future. The struct is not meant to be constructed outside of `iroh-net` anyways.

- `iroh-base`
  - removed
    - `iroh_base::rpc::RpcError`
    - `iroh_base::rpc::RpcResult`

- `iroh`
  - changed
    - `iroh::node::ProtocolHandler` -> `iroh::router::ProtocolHandler`

## 🐛 Bug Fixes

- Switch to correctly patched quic-rpc and iroh-quinn - ([d925da4](https://github.com/n0-computer/iroh/commit/d925da442993fb79d55b905d4c17a324e9549bd2))

## 📚 Documentation

- Fixup changelog - ([5066102](https://github.com/n0-computer/iroh/commit/50661022258e607775af6e6b83c4c25fc57ed088))

## ⚙️ Miscellaneous Tasks

- Release - ([134a93b](https://github.com/n0-computer/iroh/commit/134a93b5a60103b3ce8fa4aacb52cdbcb291d00b))

**Comment by crimeminister** [comment]
If the iroh CLI is deprecated, is the intended usage of iroh from here on solely as a library? Would you discourage a third party from developing a new CLI or adopting the old one?

**Reply by matheus23** [reply]
Third parties should totally build CLIs!

In fact, we may go this route with some of our own protocols.
Instead of one giant "iroh CLI" that can do everything, we *might* develop a CLI for iroh-docs or iroh-willow.

Similarly, we expect others to build CLIs for protocols where appropriate.
